iib/workers/tasks/opm_operations.py

```diff
@@ -918,3 +922,259 @@ def verify_operator_exists(
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+
+    log.info(
+        'Generating the database file from an existing database %s and excluding'
+        ' the following operator(s): %s',
+        from_index,
+        ', '.join(operators),
+    )
+
+    with set_registry_token(overwrite_from_index_token, from_index, append=True):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to remove operators from the index image')
+
+
+def deprecate_bundles(
+    bundles: List[str],
+    base_dir: str,
+    binary_image: str,
+    from_index: str,
+    overwrite_target_index_token: Optional[str] = None,
+    container_tool: Optional[str] = None,
+) -> None:
+    """
+    Deprecate the specified bundles from the index image.
+
+    Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required to use ``overwrite_target_index``.
+        The format of the token must be in the format "user:password".
+    :param str container_tool: the container tool to be used to operate on the index image
+    """
+    from iib.workers.tasks.utils import run_cmd, set_registry_token
+
+    cmd = [
+        opm_path,
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+        '--allow-package-removal',
+    ]
+    if container_tool:
+        cmd.append('--container-tool')
+        cmd.append(container_tool)
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles')
+
+
+def set_opm(func: Callable) -> Callable:
+    """
+    Set opm version for each IIB operation.
+
+    :param function func: the function to be decorated.
+    :return: the decorated function
+    :rtype: function
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        from iib.workers.tasks.utils import _get_function_arg_value
+
+        if 'from_index' in inspect.getfullargspec(func).args:
+            from_index = _get_function_arg_value('from_index', func, args, kwargs)
+        # if target_index is present instead of from_index
+        if 'target_index' in inspect.getfullargspec(func).args:
+            from_index = _get_function_arg_value('target_index', func, args, kwargs)
+        try:
+            set_opm_verison(from_index)
+            func(*args, **kwargs)
+        finally:
+            global opm_path
+            opm_path = 'opm'
```

Comment:
I am not sure if I understand, but will this line overwrite the previously set opm path in the set_opm_version()?

Reply:
Yes, we want the during the entire handle_add/rm request a specific opm should be used, and then after the function completes, it should be set back to default